tests/phpunit/tests/post/wpTrashPost.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+class Tests_Post_wpTrashPost extends WP_UnitTestCase {
+	/**
+	 * @var WP_Post
+	 */
+	protected $post;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->post = $this->factory()->post->create_and_get(
+			array(
+				'post_status' => 'draft',
+			)
+		);
+	}
+
+	public function test_trash_post() {
```

Comment:
Thanks for these tests @Hug0-Drelon!

These should ideally be separate test methods to test each filter/action independently. That way, if there's a failure with one of them, the others will still be tested. Same for the `wp_untrash_post()` tests.